storage/buckets/buckets_test.go

```diff
@@ -278,22 +279,29 @@ func TestLifecycleManagement(t *testing.T) {
 	tc := testutil.SystemTest(t)
 	bucketName := tc.ProjectID + "-storage-buckets-tests"
 
+	ctx := context.Background()
+	testutil.CleanBucket(ctx, t, tc.ProjectID, bucketName)
+
 	if err := enableBucketLifecycleManagement(ioutil.Discard, bucketName); err != nil {
 		t.Errorf("enableBucketLifecycleManagement: %v", err)
 	}
 
 	// verify lifecycle is set
-	m, err := getBucketMetadata(ioutil.Discard, bucketName)
+	client, err := storage.NewClient(ctx)
 	if err != nil {
-		t.Errorf("getBucketMetadata: %#v", err)
+		t.Errorf("storage.NewClient: %v", err)
+	}
+	attrs, err := client.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		t.Errorf("Bucket(%q).Attrs: %v", bucketName, err)
 	}
 
 	want := storage.LifecycleRule{
 		Action:    storage.LifecycleAction{Type: "Delete"},
 		Condition: storage.LifecycleCondition{AgeInDays: 100},
 	}
 
-	if r := m.Lifecycle.Rules[0]; !reflect.DeepEqual(r, want) {
+	if r := attrs.Lifecycle.Rules[0]; !reflect.DeepEqual(r, want) {
```

Comment:
If attr.Lifecycle.Rules is empty, this will panic. I would check the length first (we would want to know that there is exactly 1 rule) so we can cause a test failure instead.